pytorch_lightning/trainer/training_io.py

```diff
@@ -316,6 +320,10 @@ def dump_checkpoint(self):
 
         checkpoint['state_dict'] = model.state_dict()
 
+        # restore native amp scaling
+        if self.use_amp and self.use_native_amp and 'native_amp_scaling_state' in checkpoint:
```

Comment:
@mcarilli sanity check this saving?

Reply:
`state_dict` is a callable, as for modules and optimizers, so `checkpoint['native_amp_scaling_state'] = self.scaler.state_dict()` is what you want.
`checkpoint['native_amp_scaling_state'] = self.scaler.state_dict` would stash the bound-method object itself :P

Reply:
Also you should make sure `state_dict()` is retrieved either at the very beginning of an iteration (before any `scaler` method calls) or at the very end (after `scaler.update()`), and that the model and optimizer state dicts are saved at that same spot.

Reply:
i thought it was a property haha. and i can’t find the docs to see if it’s a method ;)

Reply:
lol i see.  it's consistent with the rest

Reply:
Another thing to consider is that with `torch.cuda.amp`, it's permissible to
- load a checkpoint from a model + optimizer not trained with Amp, and resume training with Amp enabled, or
- load a checkpoint from a model + optimizer trained with Amp, and resume training without Amp.

I think your `if` criteria are flexible enough that that may naturally happen with the right user args but I'm not sure just from looking at it.

Reply:
yeah this code works.

## Case 1: Train with amp, load amp
works fine

## case 2: Train amp, load and not use amp
in this case, lightning loads the amp state but amp is disabled so user doesn't use it at all

## case 3: train regular, resume regular
works fine

## case 4: train regular, resume with amp
in this case the checkpoint has no amp state and model starts normal but on amp.